### PR TITLE
Support for lineHeight in DOM Renderer

### PR DESCRIPTION
**Pull request by leomoty**
Related to #1700. I've attempted to port the calculations from Canvas renderer, it seems to be a few pixels off.

Also, does xterm.js support zooming the browser officially? That is not working properly.

@Tyriar, can you please take a look.

### CODE REVIEW

**Comment by Tyriar** [comment]
> Also, does xterm.js support zooming the browser officially? That is not working properly.

@leomoty yes using the browser zoom should work.

**Comment by Tyriar** [comment]
Hm, seems alright to me. Note that you're not aiming for exact parity with the canvas renderer, it's fine if they're a little different.

**Comment by leomoty** [comment]
@Tyriar, updated `_createSelectionElement`, the selection height is working properly now, but the selection doesn't seem to match as you go down the lines. Any pointers?

**Comment by Tyriar** [comment]
@leomoty here's the function that grabs the coordinates from the MouseEvent:

https://github.com/xtermjs/xterm.js/blob/b650db5b1ae230d4c7adeb39c433be28712dc66f/src/SelectionManager.ts#L485

**Comment by leomoty** [comment]
@Tyriar I have been a bit busy with work, just noticed the status change, I also took another look, I might have found the issue. Unsure why it happens tho.

Following the code you gave me it goes here:

https://github.com/xtermjs/xterm.js/blob/master/src/utils/MouseHelper.ts#L51-L72

When I tried to inspect `this._renderer.dimensions` itself, it doesnt seem to account for the changes I did in: 

https://github.com/xtermjs/xterm.js/pull/1733/files#diff-0a9d734ca7737653ca5c980f3c663f7aR96

I.e. `actualCellHeight` is 17px instead of the expected 34px. I can confirm that inside `_updateDimensions` it is fine.

**Comment by leomoty** [comment]
As a follow-up: `getCoords` gets lineHeight passed in, if I apply that to `actualCellHeight` it works as expected.

 I wonder if this might break the canvas renderer tho. I'll wait for your feedback.

**Comment by leomoty** [comment]
@Tyriar Yeah, I didn't mean to change canvas renderer nor selection at all, when I noticed the `MouseHelper@getCoords` function, I checked the `this._renderer` object there.

What I noticed is:
- Canvas renderer works fine
- The changes I did to Dom Renderer are not applied

Inside `DomRenderer@_updateDimensions` with `lineHeight` = 2.0, I get 34px for `actualCellHeight`.
However `MouseHelper@getCoords` still accounts for 17px.

I have not been able to track down why it happens.

File: src/Terminal.ts [hunk 1]
```diff
@@ -473,6 +473,7 @@ export class Terminal extends EventEmitter implements ITerminal, IDisposable, II
         if (this._theme) {
           this.renderer.setTheme(this._theme);
         }
+        this.mouseHelper = new MouseHelper(this.renderer);
```
Comment (leomoty) [inline src/Terminal.ts]: @Tyriar when `lineHeight` is updated,  it falls through to `rendererType`, and thus destroying the renderer, in which mouseHelper is never updated.

I am not sure we want to destroy the current renderer when those props are updated.

Either way, selection works properly with this change.

**Pull request merged**